dev/make/function_definitions/32e.mk

```diff
@@ -0,0 +1,108 @@
+#===============================================================================
```

Comment:
Should we rename this to be aligned with ARM? 

This is quite a legacy name as one of variants between 32bit, em64t and itanium64 variants.  

Any ideas on what this would be?  basically this is generic x86 architecture

Reply:
how about x86_64.mk?

Reply:
Agreed that this could be a better name, and x86_64 sounds decent. However, changing this requires updates outside of the make setup (there are ci scripts that depend on this value), and I think it would be better done as another merge request to change the name. What do you think? I'm happy to put something up for that after this.

Reply:
yes, x86_64 looks decent. - the only thing to check is _ handling and this would depend on how wide we would like to have this implemented, but in longerterm worth having better name.

yes, it's fine to keep this separate as renaming would require updates for internal CI. @homksei @ethanglaser  - this is something that we can look on and rename legacy 32e naming to have better match/separation in future.

Reply:
https://github.com/oneapi-src/oneDAL/pull/2674 as a draft merge request, for what this would look like on top of this commit.

Is there anything else that needs changing on my end for this merge request? I can see that all the public CI passes, but the Intel CI is failing